server/pairing/sync_device_test.go

```diff
@@ -0,0 +1,316 @@
+	require.NoError(s.T(), err)
+
+	walletDerivedAccount := derivedAddresses[pathDefaultWallet]
+	walletAccount := &accounts.Account{
+		PublicKey: types.Hex2Bytes(walletDerivedAccount.PublicKey),
+		KeyUID:    generatedAccountInfo.KeyUID,
+		Address:   types.HexToAddress(walletDerivedAccount.Address),
+		Color:     "",
+		Wallet:    true,
+		Path:      pathDefaultWallet,
+		Name:      "Ethereum account",
+	}
+
+	chatDerivedAccount := derivedAddresses[pathDefaultChat]
+	chatAccount := &accounts.Account{
+		PublicKey: types.Hex2Bytes(chatDerivedAccount.PublicKey),
+		KeyUID:    generatedAccountInfo.KeyUID,
+		Address:   types.HexToAddress(chatDerivedAccount.Address),
+		Name:      settings.Name,
+		Chat:      true,
+		Path:      pathDefaultChat,
+	}
+
+	accounts := []*accounts.Account{walletAccount, chatAccount}
+	err = backend.StartNodeWithAccountAndInitialConfig(account, s.password, *settings, nodeConfig, accounts)
+	require.NoError(s.T(), err)
+	return backend
+}
+
+func (s *SyncDeviceSuite) prepareBackendWithoutAccount(tmpdir string) *api.GethStatusBackend {
+	backend := api.NewGethStatusBackend()
+	backend.UpdateRootDataDir(tmpdir)
+	return backend
+}
+
+func (s *SyncDeviceSuite) TestPairingSyncDeviceClientAsSender() {
+	clientTmpDir := filepath.Join(s.clientAsSenderTmpdir, "client")
+	clientBackend := s.prepareBackendWithAccount(clientTmpDir)
+
+	serverTmpDir := filepath.Join(s.clientAsSenderTmpdir, "server")
+	serverBackend := s.prepareBackendWithoutAccount(serverTmpDir)
+	defer func() {
+		require.NoError(s.T(), serverBackend.Logout())
+	}()
+
+	err := serverBackend.AccountManager().InitKeystore(filepath.Join(serverTmpDir, keystoreDir))
+	require.NoError(s.T(), err)
+	err = serverBackend.OpenAccounts()
+	require.NoError(s.T(), err)
+	serverKeystorePath := filepath.Join(serverTmpDir, keystoreDir)
+	configJSON := fmt.Sprintf(`{"KeystorePath":"%s"}`, serverKeystorePath)
+	cs, err := StartUpPairingServer(serverBackend, Receiving, configJSON)
+	require.NoError(s.T(), err)
+
+	// generate some data for the client
+	clientBrowserAPI := clientBackend.StatusNode().BrowserService().APIs()[0].Service.(*browsers.API)
+	_, err = clientBrowserAPI.StoreBookmark(context.TODO(), browsers.Bookmark{
+		Name: "status.im",
+		URL:  "https://status.im",
+	})
+	require.NoError(s.T(), err)
+
+	activeAccount, err := clientBackend.GetActiveAccount()
+	require.NoError(s.T(), err)
+	clientKeystorePath := filepath.Join(clientTmpDir, keystoreDir, activeAccount.KeyUID)
+	var config = PayloadSourceConfig{
+		KeystorePath: clientKeystorePath,
+		KeyUID:       activeAccount.KeyUID,
+		Password:     s.password,
+	}
+	configBytes, err := json.Marshal(config)
+	require.NoError(s.T(), err)
+	err = StartUpPairingClient(clientBackend, cs, string(configBytes))
+	require.NoError(s.T(), err)
+	require.NoError(s.T(), clientBackend.Logout())
+
+	serverBrowserAPI := serverBackend.StatusNode().BrowserService().APIs()[0].Service.(*browsers.API)
```

Comment:
I haven't found a better way to access browsers.API as it's too nested, if u have better way that would be appreciated